pixeltable/catalog/table_version.py

```diff
@@ -645,29 +678,32 @@ def _add_columns(
 
             plan, value_expr_slot_idx = Planner.create_add_column_plan(self.path, col)
             plan.ctx.num_rows = row_count
-
             try:
                 plan.open()
                 try:
-                    num_excs = self.store_tbl.load_column(col, plan, value_expr_slot_idx, on_error)
+                    excs_per_col = self.store_tbl.load_column(col, plan, value_expr_slot_idx, on_error)
                 except sql.exc.DBAPIError as exc:
                     # Wrap the DBAPIError in an excs.Error to unify processing in the subsequent except block
                     raise excs.Error(f'SQL error during execution of computed column `{col.name}`:\n{exc}') from exc
-                if num_excs > 0:
+                if excs_per_col > 0:
                     cols_with_excs.append(col)
-            except excs.Error as exc:
-                self.cols.pop()
-                for col in cols:
-                    # remove columns that we already added
-                    if col.id not in self.cols_by_id:
-                        continue
-                    if col.name is not None:
-                        del self.cols_by_name[col.name]
-                    del self.cols_by_id[col.id]
-                # we need to re-initialize the sqlalchemy schema
-                self.store_tbl.create_sa_tbl()
+                    num_excs += excs_per_col
+            except Exception as exc:
```

Comment:
The control flow requires an explanation (eg, why catch Exception, etc.).

Probably best to put into exception_handler.py. It should give specific instructions on how it needs to be used.

Reply:
I wonder if we even need to catch exception here, as when handler is called from finally it can detect if an exception is raised or not